Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1263,6 +1266,92 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        // Field used for accessing the ItemsSource though the interface ISupportIncrementalLoading
+        private ISupportIncrementalLoading _incrementalItemsSource;
+
+        private bool _loadingItems;
+
+        // Method that loads additional data once the user has scrolled to a certain threashold
+        private async void LoadMoreDataFromIncrementalItemsSource()
+        {
+            if (IncrementalLoadingTrigger == IncrementalLoadingTrigger.Edge && _incrementalItemsSource != null && _incrementalItemsSource.HasMoreItems && !_loadingItems)
+            {
+                var bottomScrollOffHeight = Math.Max(0, EdgedRowsHeightCalculated - CellsHeight - VerticalOffset);
+
+                var rowsPresenterAvailableHeight = this._rowsPresenterAvailableSize.HasValue
+                    ? _rowsPresenterAvailableSize.Value.Height
+                    : _vScrollBar.ViewportSize;
+
+                if ((IncrementalLoadingThreshold * rowsPresenterAvailableHeight) >= bottomScrollOffHeight)
+                {
+                    _loadingItems = true;
+
+                    var numberOfRowsToLoad = (uint)(DataFetchSize * rowsPresenterAvailableHeight / this.RowHeightEstimate);
+
```

Comment:
"The number passed to LoadMoreItemsAsync is at least equal to 1."

Reply:
Aren't we lucky to have this.RowHeightEstimate, @filipwa84?   Thanks for working on this feature.

Reply:
Ah, forgot about that. Indeed very handy :)